src/main/java/org/opensearch/neuralsearch/processor/NeuralSparseTwoPhaseProcessor.java

```diff
@@ -191,11 +204,54 @@ private RescorerBuilder<QueryRescorerBuilder> buildRescoreQueryBuilderForTwoPhas
         return twoPhaseRescorer;
     }
 
+    @SuppressWarnings("unchecked")
+    private void validateSeismicQuery(String[] indices, Multimap<NeuralSparseQueryBuilder, Float> queryBuilderMap) {
+        for (String index : indices) {
+            final IndexMetadata metadata = clusterService.state().metadata().index(index);
+            if (metadata == null || !SparseSettings.IS_SPARSE_INDEX_SETTING.get(metadata.getSettings())) {
+                return;
+            }
+            MappingMetadata mappingMetadata = metadata.mapping();
+            if (mappingMetadata == null || mappingMetadata.sourceAsMap() == null) {
+                return;
+            }
+            Object properties = mappingMetadata.sourceAsMap().get("properties");
+            if (!(properties instanceof Map)) {
+                return;
+            }
+            Set<String> sparseAnnFields = new HashSet<>();
+            Map<String, Object> fields = (Map<String, Object>) properties;
+            for (Map.Entry<String, Object> field : fields.entrySet()) {
+                Map<String, Object> fieldMap = (Map<String, Object>) field.getValue();
+                Object type = fieldMap.get("type");
+                if (Objects.nonNull(type) && SparseTokensFieldType.isSparseTokensType(type.toString())) {
+                    sparseAnnFields.add(field.getKey());
+                }
```

Comment:
can you put this logic into a common utility class as `getSparseAnnFields` that later I can modify the token id PR to use it.